### PR TITLE
Feature/cbsfpksk generation

### PR DESCRIPTION
**Pull request by IceTDrinker**
### Description

Engine to easily generate the fpksk required for a circuit bootstrap

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [x] The draft release description has been updated
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Comment by IceTDrinker** [comment]
forgot to add checks for the generic error stuff, don't review yet, sorry, not sure how I missed that

**Review by agnesLeroy — COMMENTED** [review]
Hello! Thanks a lot for this PR! In the end you didn't add a fixture+bench for this one, maybe it'd be worth to do it? Regarding the documentation, would it be possible to add a formal definition for the engine (I can't remember if the entity had one). If it's complicated we need to open an issue for it.

**Comment by IceTDrinker** [comment]
Sorry this PR was not in a state to review, I'm finishing up some stuff and I was going to follow up with the fixtures

File: concrete-core/src/specification/engines/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys_generation.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+engine_error! {
+    LweCircuitBootstrapPrivateFunctionalPackingKeyswitchKeysGenerationError for
+    LweCircuitBootstrapPrivateFunctionalPackingKeyswitchKeysGenerationEngine @
+}
+
+impl<EngineError: std::error::Error>
+    LweCircuitBootstrapPrivateFunctionalPackingKeyswitchKeysGenerationError<EngineError>
+{
+    /// Validates the inputs
```
Comment (agnesLeroy) [inline concrete-core/src/specification/engines/lwe_circuit_bootstrap_private_functional_packing_keyswitch_keys_generation.rs:18]: (I let you add the generic checks, just re-request my review when you've pushed :wink:)

**Comment by IceTDrinker** [comment]
updated with fixtures and generic checks on engine, commit by commit should be more readable

**Review by agnesLeroy — APPROVED** [review]
Looks good to me! Thanks a lot!

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Pull request merged**